app/controllers/tasks_controller.rb

```diff
@@ -1,80 +1,133 @@
+    if @task[:completed_at].nil?
+      Task.completed(params[:id])
+    else
+      Task.not_completed(params[:id])
+    end
+
+    redirect_to "/"
   end
 
   def show
-    task_instance_vars
-    @visibility = "invisible"
+    task = Task.find(params[:id])
+
+    @task_id =              task[:id]
+    @task_name =            task[:name]
+    @task_description =     task[:description]
+    @task_status =          task[:completed_at].nil? ? "NEED TO PLAY" : "FINISHED"
+    @task_completion_date = task[:completed_at].nil? ? "N/A" : date_format(task[:completed_at])
+    @task_created_at =      date_format(task[:created_at])
+    @task_updated_at =      date_format(task[:updated_at])
+    @task_recommended_id =  task.person.id
+    @task_recommended_by =  task.person.name
+
+    @button_visibility = "visible"
+    @delete_visibility = "invisible"
 
     render :task
+      # NOTE TO SELF: "render 'task'" also would work
   end
 
   def show_before_delete
     task_instance_vars
-    @visibility = "visible"
+    @button_visibility = "invisible"
+    @delete_visibility = "visible"
 
     render :task
   end
 
   def new
     @task = Task.new
+    @person = Person.new
 
     render :new
   end
 
   def create
-    @task = Task.new(create_params[:task])
+    @task = Task.new(permit_params[:task])
     @task.save
 
     redirect_to "/" # Why does this not work with :index?
       # NOTE!!!: Look up other ways to write 'redirect_to'.
   end
 
+  def edit
+    task_instance_vars
+
+    render :edit
+  end
+
+  def update
+    # raise
+    completed = params[:task][:completed_at] || Task.find(params[:id]).completed_at
+    Task.editing(params[:id], params[:task], completed)
+
+    ### QUESTION: Why is it, when I use a method from the model it doesn't need #permit?
+    ### But when editing from the controller, you need to permit the params?
+    # @task = Task.find(params[:id])
+    # @task.attributes = {
+    #   name: edit_params[:task][:name],
+    #   description: edit_params[:task][:description]
+    # }
+    # @task.save
+
+    redirect_to "/tasks/#{params[:id]}"
+  end
+
   def destroy
     # @task = Task.find(params[:id])
-    Task.find(params[:id]).destroy
+    Task.destroy_one(params[:id])
 
     redirect_to "/"
   end
 
   def destroy_all
-    @all_completed_tasks = Task.where.not(completed_at: nil)
+    @all_completed_tasks = Task.all_completed
 
     render :delete_all
   end
 
   def destroy_all_completed
-    completed_tasks = Task.where.not(completed_at: nil)
-    completed_tasks.destroy_all
+    Task.destroy_all_completed
 
     redirect_to "/"
   end
 
-  private
 
-  def completed_instance_var
-    # QUESTION: Constant's don't really work, do they? (Didn't work for me.)
-    @completed = "\u2713"
-    @uncompleted = ""
-    # What is this?
-    # "\u2713".force_encoding("UTF-8")
-  end
+  private
 
   def task_instance_vars
     # NOTE!!! Need to read up on params to show unique URLs!
     @task = Task.find(params[:id])
```

Comment:
I think you could remove all of this extra logic and just use the `@task` instance variable